data_processing.py

```diff
@@ -44,3 +45,16 @@ def load_file_to_corpus(filepath, corpus):
         corpus[key] = f.read()
 
 
+def pickle_save_data_to_file(data, path):
+    dir, _ = os.path.split(path)
+    if not os.path.exists(dir):
+        os.makedirs(dir)
```

Comment:
Maybe this is nicer:
```
path = Path(path)
path.mkdir(parents=True, exist_ok=True)
```

Reply:
Will do it!

Reply:
This is raising an error when the file already exists (or we may want to override existing file). I will leave it as it is in my PR since it is working. Feel free to change it if you find a better way to do it !

Reply:
It shouldn’t return an error with the `exist_ok=True` argument`

Reply:
Actually it does when the last element of the path is an already existing file
https://docs.python.org/3/library/pathlib.html#pathlib.Path.mkdir

Reply:
Ah yes ok, missed that bit, i’ll change to `path.parents[0].mkdir(parents=True, exist_ok=True)`